packages/headless/src/features/question-answering/question-answering-insight-analytics-actions.ts

```diff
@@ -132,46 +133,45 @@ export const logOpenSmartSnippetSource = (): InsightAction =>
 export const logOpenSmartSnippetInlineLink = (
   payload: QuestionAnsweringInlineLinkActionCreatorPayload
 ): InsightAction =>
-  makeInsightAnalyticsAction(
-    'analytics/smartSnippet/source/open',
-    (client, state) => {
-      validatePayload(payload, inlineLinkPayloadDefinition());
-      const result = answerSourceSelector(state)!;
-      return client.logOpenSmartSnippetInlineLink(
-        partialDocumentInformation(result, state),
-        {
-          ...documentIdentifier(result),
-          ...payload,
-        },
-        getCaseContextAnalyticsMetadata(state.insightCaseContext)
-      );
-    }
-  );
+  makeInsightAnalyticsActionFactory(
+    SearchPageEvents.openSmartSnippetInlineLink
+  )('analytics/smartSnippet/source/open', (client, state) => {
+    validatePayload(payload, inlineLinkPayloadDefinition());
+    const result = answerSourceSelector(state)!;
```

Comment:
similar to https://github.com/coveo/ui-kit/pull/3726/files#diff-3e38ed2b0d0e97f7e8fb3c19ce6b0894356c5b7cd14b81e8313bacf1275f5e89R81

if the selector doesn't return a result, should we simply return without logging the event instead of throwing a TypeError at runtime?

Reply:
Sure we could!

Reply:
the current behaviour if the result is not found 2 is that no event will be sent but no error will be thrown, I think we could have a discussion about the utility of having a smart snippet component in page 2 at the first place.
But this out of the scoope of this PR.